### PR TITLE
setup: Next, TS, tailwind, prisma, next-auth, shadcn

### LINKED ISSUE #18: [ FEATURE ] : ⭐ Intializing the Repo
#### Description
We need to set up the initial repository with the following technologies: Next.js, Prisma, Shadcn, Tailwind CSS, and TypeScript. Additionally, we require a well-designed landing page and an authentication setup. Bonus points for including a job ad posting feature.

#### Goals
1. Initialize the repository with Next.js, Prisma, Shadcn, Tailwind CSS, and TypeScript.
2. Create a visually appealing landing page.
3. Implement user authentication.
4. (Optional) Add a job ad posting feature.

#### Steps
1. **Initialize the Project**
   - Set up a new Next.js project.
   - Integrate TypeScript into the Next.js project.
   - Add Tailwind CSS for styling.
   - Integrate Prisma for database management.
   - Add Shadcn components.

2. **Create the Landing Page**
   - Design and implement a responsive landing page.
   - Ensure the landing page includes sections for the main features and a call to action.

3. **Set Up Authentication**
   - Implement authentication using a suitable library (e.g., NextAuth.js).
   - Include login, registration, and logout functionalities.
   - Ensure user sessions are managed securely.

4. **(Optional) Job Ad Posting Feature**
   - Implement functionality for users to post job ads.
   - Create a form for job ad submission.
   - Display submitted job ads on a separate page or section.

#### Acceptance Criteria
- [ ] The repository is initialized with Next.js, Prisma, Shadcn, Tailwind CSS, and TypeScript.
- [ ] A well-designed landing page is implemented and responsive.
- [ ] Authentication setup is completed with login, registration, and logout functionalities.
- [ ] (Optional) Users can post job ads, and these ads are displayed correctly.

#### Additional Notes
- Ensure the code follows best practices and is well-documented.
- Tests should be included where applicable.
- Provide clear instructions on how to set up and run the project locally.

By completing this feature, we will have a solid foundation for our project, making it easier to add new features in the future. Contributions that include the optional job ad posting feature will be given priority for merging.

### PR DESCRIPTION
**Pull request by Rishabh-Singh-Codes**
Created initial setup with required tech stack.

#### Acceptance Criteria
- [x] The repository is initialized with Next.js, Prisma, Shadcn, Tailwind CSS, and TypeScript.
- [x] A well-designed landing page is implemented and responsive.
- [x] Authentication setup is completed with login, registration, and logout functionalities.
- [ ] (Optional) Users can post job ads, and these ads are displayed correctly.

#### Preview
Screenshot 2024-08-05 at 2 11 30 PM

https://github.com/user-attachments/assets/66598e1c-2702-44dd-8a97-0a1f63336c5f

Closes #18

### CODE REVIEW

**Comment by Rishabh-Singh-Codes** [comment]
@VineeTagarwaL-code

File: app/api/auth/[...nextauth]/route.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const handler = NextAuth({
+  providers: [
+    GoogleProvider({
+      clientId: process.env.GOOGLE_CLIENT_ID ?? "",
+      clientSecret: process.env.GOOGLE_CLIENT_SECRET ?? "",
+    }),
+  ],
+  callbacks: {
+    async signIn({ user, account, profile, email, credentials }) {
+      console.log(user, account, profile, email, credentials);
```
Comment (VineeTagarwaL-code) [inline app/api/auth/[...nextauth]/route.ts:14]: remove console.log

File: app/page.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      <h2 className="text-xl font-semibold">
+        An initiative by{" "}
+        <a href="https://100xdevs.com/" className="font-bold underline">
+          100xDevs
+        </a>
+      </h2>
+
+      <div className="border w-3/5 min-h-96 mt-12 rounded-lg">
+      <h1 className="text-center my-6 font-bold text-2xl">Latest Jobs</h1>
+      <div className="border my-4 mx-10 rounded-lg min-h-20 animate-pulse bg-slate-100"></div>
```
Comment (VineeTagarwaL-code) [inline app/page.tsx:15]: what are these for ?

File: components/Appbar.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+"use client";
+import { signIn, signOut, useSession } from "next-auth/react";
+import { Button } from "./ui/button";
+
+export const Appbar = () => {
+  const session = useSession();
+  console.log('session', session)
```
Comment (VineeTagarwaL-code) [inline components/Appbar.tsx:7]: Remove logs

File: readme.md [hunk 4]
```diff
@@ -1 +1 @@
-# Job board
+# Job board
```
Comment (VineeTagarwaL-code) [inline readme.md:1]: A guide on how to setup locally, explaining your env and other stuff is required

**Review by VineeTagarwaL-code — CHANGES_REQUESTED** [review]
I have added all the comments & feedback on your code .. please make sure you do the modification and add your next PR, would love to see the last task checked also !

File: app/page.tsx (hunk 2 above)
Comment (Rishabh-Singh-Codes) [reply app/page.tsx:15]: These are skeletons for showcasing where the latest jobs will be displayed

**Comment by VineeTagarwaL-code** [comment]
click on resolve conversation to close and then ping me again
image

**Comment by Rishabh-Singh-Codes** [comment]
> click on resolve conversation to close and then ping me again 

done

**Comment by VineeTagarwaL-code** [comment]
Swetanshu's PR was prioritized as it included all the mentioned features and was approved by Harkirat. Additionally, it was the first PR to be opened by him.

**Pull request closed without merging**